helix-term/src/commands/typed.rs

```diff
@@ -1034,6 +1034,56 @@ fn reload(
     })
 }
 
+fn reload_all(
+    cx: &mut compositor::Context,
+    _args: &[Cow<str>],
+    event: PromptEvent,
+) -> anyhow::Result<()> {
+    if event != PromptEvent::Validate {
+        return Ok(());
+    }
+
+    let scrolloff = cx.editor.config().scrolloff;
+    let view_id = view!(cx.editor).id;
+
+    let docs_views: Vec<(DocumentId, Vec<ViewId>)> = cx
+        .editor
+        .documents_mut()
+        .map(|doc| {
+            let mut views: Vec<_> = doc.selections().keys().cloned().collect();
+
+            if views.is_empty() {
+                doc.ensure_view_init(view_id);
+                views = vec![view_id];
+            };
+
+            (doc.id(), views)
+        })
+        .collect();
+
+    for (doc_id, view_ids) in docs_views {
+        let doc = doc_mut!(cx.editor, &doc_id);
+
+        let mut view_ids = view_ids.into_iter();
+
+        // Every doc is guaranteed to have at least 1 view at this point.
+        let view_id = view_ids.next().unwrap();
+        let view = view_mut!(cx.editor, view_id);
+
+        doc.reload(view).map(|_| {
+            view.ensure_cursor_in_view(doc, scrolloff);
+        })?;
+
+        for view_id in view_ids {
+            let view = view_mut!(cx.editor, view_id);
+
+            view.ensure_cursor_in_view(doc, scrolloff);
+        }
```

Comment:
This can be simplified a bit:

```suggestion
	// Every doc is guaranteed to have at least 1 view at this point.
        let view = view_mut!(cx.editor, view_ids[0]);
        doc.reload(view)?;

        for view_id in view_ids {
            let view = view_mut!(cx.editor, view_id);

            view.ensure_cursor_in_view(doc, scrolloff);
        }
```